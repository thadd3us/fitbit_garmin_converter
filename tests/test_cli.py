import subprocess
from pathlib import Path

def test_convert_weight_cli(tmp_path, snapshot):
    output_file = tmp_path / "output.csv"
    test_data_dir = Path(__file__).parent / "test_data"
    
    result = subprocess.run([
        "python", "-m", "fitbit_garmin_converter.cli",
        str(test_data_dir), "--output-file", str(output_file)
    ], capture_output=True, text=True, cwd=Path(__file__).parent.parent)
    
    assert result.returncode == 0
    assert snapshot == output_file.read_text()