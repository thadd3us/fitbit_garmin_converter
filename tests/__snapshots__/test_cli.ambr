# serializer version: 1
# name: test_convert_weight_cli
  '''
  Date,Weight,BMI,Fat
  2025-08-09T19:21:54,190.9,25.86,0
  2025-08-11T06:28:07,190.6,25.83,0
  2025-08-12T07:30:00,189.2,25.64,0
  
  '''
# ---
